### ISSUE TITLE
Add PTS/DTS continuity and monotonicity check

### ISSUE THREAD

**Issue by small-teton** [small-teton/mpeg-ts-analyzer#22]
## Overview

Detect PTS/DTS anomalies that cause playback issues.

## Background

Properly incrementing PTS/DTS values are essential for smooth playback. In practice, various problems can occur:

- **Non-monotonic timestamps**: PTS/DTS goes backward unexpectedly, causing frame reordering issues at the decoder
- **Wraparound**: The 33-bit PTS/DTS counter overflows (every ~26.5 hours). Decoders should handle this, but many don't gracefully
- **Unexpected gaps**: Large jumps in timestamps (e.g., after ad insertion or stream splicing) can cause buffering or freeze
- **PTS/DTS relationship violations**: DTS must be ≤ PTS for the same access unit; DTS of consecutive frames in decode order must be non-decreasing

These issues are common in live streaming (especially with ad insertion, stream switching, and redundancy failover) and are notoriously hard to diagnose without packet-level inspection.

## Proposed Output

```
Timestamp Anomaly Report:
  PID 0x0100: PTS non-monotonic at 0x00123456 (2041.977ms -> 2001.977ms)
  PID 0x0100: PTS gap of 5.2s at 0x00234567 (expected ~40ms)
  PID 0x0101: PTS wraparound at 0x00345678
```

## Design

**Always enabled** — Timestamps are already parsed in the existing flow. Adding anomaly checks requires minimal computation and the output is only printed when problems are detected, so there is no noise for healthy streams.

## Use Cases

- Debugging A/V sync issues in live streams
- Validating ad insertion / stream splicing output
- Verifying encoder output over long-running streams (wraparound handling)
